shell/lisa_shell

```diff
@@ -396,7 +396,7 @@ EOF
 
     # Check for required external dependencies
 
-    if which realpath &>/dev/null; then
+    if [[ -z $(which realpath) ]]; then
```

Comment:
If we are not interested in stderr, that form can result in a slightly shorter/less clunky line:
```
if not which realpath &>/dev/null; then 
```

Reply:
Seeing as we have `-z`/ `-x` checks everywhere, I figured I'd try to keep this in line.
Also, that `not` command isn't standard, is it?

Reply:
Right, `not` seems to be provided by llvm package on my system, but `!`  definitely is [1]. It only checks the return code of the command, which is exactly what we need here (taken from `man which`):
```
RETURN VALUE
       Which returns the number of failed arguments, or -1 when no `program‐
       name' was given.
```
`-z` checks if a string is empty, which seems equivalent here.

[1]  "2.9.2 Pipelines" in http://pubs.opengroup.org/onlinepubs/009696899/utilities/xcu_chap02.html